source/jormungandr/jormungandr/street_network/with_parking.py

```diff
@@ -0,0 +1,114 @@
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import logging
+from jormungandr.street_network.street_network import AbstractStreetNetworkService
+from jormungandr import street_network
+
+from jormungandr.street_network.asgard import Asgard
+from jormungandr import utils
+from navitiacommon import response_pb2
+
+
+class WithParking(AbstractStreetNetworkService):
+    def __init__(self, instance, service_url, modes, id=None, timeout=10, api_key=None, **kwargs):
+        self.instance = instance
+        self.modes = modes or []
+        self.sn_system_id = id or 'with_parking'
+        self.parking_module = utils.create_object(kwargs.get('parking', None))
+        config = kwargs.get('street_network', None)
+        if 'service_url' not in config['args']:
+            config['args'].update({'service_url': None})
+        if 'instance' not in config['args']:
+            config['args'].update({'instance': instance})
+
+        self.street_network = utils.create_object(config)
+
+    def status(self):
+        return {'id': unicode(self.sn_system_id), 'class': self.__class__.__name__, 'modes': self.modes}
+
+    def _direct_path(
+        self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+    ):
+        response = self.street_network._direct_path(
+            mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+        )
+        if response and len(response.journeys):
+            self.add_parking_section_in_direct_path(response)
+
+        return response
+
+    def add_parking_section_in_direct_path(self, response):
+        logger = logging.getLogger(__name__)
+        logger.debug("Creating parking section for direct path")
+        for journey in response.journeys:
+            section = journey.sections.add()
+            section.duration = self.parking_module.park_duration
+            section.type = response_pb2.PARK
+            section.id = 'section_1'
+            section.begin_date_time = journey.sections[0].end_date_time
+            section.end_date_time = section.begin_date_time + section.duration
+            journey.arrival_date_time = section.end_date_time
```

Comment:
you forgot `journey.duration += self.parking_module.park_duration`( you've done in the last commit :) )

Reply:
Nice catch ! Just added it :)